src/main/java/uk/gov/hmcts/reform/iacasedocumentsapi/domain/templates/letter/InternalDetainedDecisionsAndReasonsDismissedLetterTemplate.java

```diff
@@ -0,0 +1,63 @@
+import static uk.gov.hmcts.reform.iacasedocumentsapi.domain.entities.AsylumCaseDefinition.APPELLANT_GIVEN_NAMES;
+import static uk.gov.hmcts.reform.iacasedocumentsapi.domain.entities.AsylumCaseDefinition.HOME_OFFICE_REFERENCE_NUMBER;
+import static uk.gov.hmcts.reform.iacasedocumentsapi.domain.utils.DateUtils.formatDateForNotificationAttachmentDocument;
+
+import java.time.LocalDate;
+import java.util.HashMap;
+import java.util.Map;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
+import uk.gov.hmcts.reform.iacasedocumentsapi.domain.entities.AsylumCase;
+import uk.gov.hmcts.reform.iacasedocumentsapi.domain.entities.ccd.CaseDetails;
+import uk.gov.hmcts.reform.iacasedocumentsapi.domain.templates.DocumentTemplate;
+import uk.gov.hmcts.reform.iacasedocumentsapi.infrastructure.CustomerServicesProvider;
+
+@Component
+public class InternalDetainedDecisionsAndReasonsDismissedLetterTemplate implements DocumentTemplate<AsylumCase> {
+
+
+    private final String templateName;
+    private final CustomerServicesProvider customerServicesProvider;
+    private final int ftpaDueCalendarDays;
+
+    public InternalDetainedDecisionsAndReasonsDismissedLetterTemplate(
+            @Value("${internalDetainedDecisionsAndReasonsDismissedLetter.templateName}") String templateName,
+            CustomerServicesProvider customerServicesProvider,
+            @Value("${internalDetainedDecisionsAndReasonsDismissedLetter.ftpaDueCalendarDays}") int ftpaDueCalendarDays
+
+    ) {
+        this.templateName = templateName;
+        this.customerServicesProvider = customerServicesProvider;
+        this.ftpaDueCalendarDays = ftpaDueCalendarDays;
+    }
+
+    public String getName() {
+        return templateName;
+    }
+
+    public Map<String, Object> mapFieldValues(
+            CaseDetails<AsylumCase> caseDetails
+    ) {
+        final AsylumCase asylumCase = caseDetails.getCaseData();
+
+        final Map<String, Object> fieldValues = new HashMap<>();
+
+        fieldValues.put("hmcts", "[userImage:hmcts.png]");
+        fieldValues.put("customerServicesTelephone", customerServicesProvider.getInternalCustomerServicesTelephone(asylumCase));
+        fieldValues.put("ADAemail", customerServicesProvider.getInternalCustomerServicesEmail(asylumCase));
+        fieldValues.put("appealReferenceNumber", asylumCase.read(APPEAL_REFERENCE_NUMBER, String.class).orElse(""));
+        fieldValues.put("homeOfficeReferenceNumber", asylumCase.read(HOME_OFFICE_REFERENCE_NUMBER, String.class).orElse(""));
+        fieldValues.put("dateLetterSent", formatDateForNotificationAttachmentDocument(LocalDate.now()));
+
+        fieldValues.put("appellantGivenNames", asylumCase.read(APPELLANT_GIVEN_NAMES, String.class).orElse(""));
+        fieldValues.put("appellantFamilyName", asylumCase.read(APPELLANT_FAMILY_NAME, String.class).orElse(""));
```

Comment:
Potential refactor here to use AsylumCaseUtils.java:getAppellantPersonalisation() same as done in InternalDetainedDecisionsAndReasonsAllowedLetterTemplate.java. Not blocking approval